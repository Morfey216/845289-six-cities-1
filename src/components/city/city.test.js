import React from 'react';
import renderer from 'react-test-renderer';
import City from './city';

const mock = {
  name: `Amsterdam`,
  isActive: true,
  onCityClick: jest.fn()
};

it(`City correctly renders with true active city`, () => {
  const tree = renderer
    .create(<City
      cityName={mock.name}
      isActive={mock.isActive}
      onCityClick={mock.onCityClick}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`City correctly renders with false active city`, () => {
  const tree = renderer
      .create(<City
        cityName={mock.name}
        isActive={!mock.isActive}
        onCityClick={mock.onCityClick}
      />)
      .toJSON();

  expect(tree).toMatchSnapshot();
});
