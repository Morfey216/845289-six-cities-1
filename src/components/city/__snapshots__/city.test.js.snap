// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City correctly renders with false active city 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item "
    onClick={[Function]}
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;

exports[`City correctly renders with true active city 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    onClick={[Function]}
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;
